### ISSUE TITLE
INSTALL ERROR

### ISSUE THREAD

**Issue by yangxiaofeill** [sestaton/tephra#26]
Dear, I want to use tephra, however, when I installed it, there are some error message as followings. Could you help me to address them?

[root@fat01 tephra]# export TEPHRA_DIR=/opt/software/tephra_fat/tephra
[root@fat01 tephra]# perl Makefile.PL
Warning: prerequisite App::Cmd 0 not found.
Warning: prerequisite List::UtilsBy 0 not found.
Warning: prerequisite Number::Range 0 not found.
Warning: prerequisite Parallel::ForkManager 0 not found.
Warning: prerequisite Set::IntervalTree 0 not found.
Warning: prerequisite Sort::Naturally 0 not found.
Warning: prerequisite Statistics::Descriptive 0 not found.
Warning: prerequisite autodie 0 not found.
Generating a Unix-style Makefile
Writing Makefile for Tephra
Writing MYMETA.yml and MYMETA.json
=====> Configuring Tephra, this may take some time......stopwatch.c: In function ‘format_time_string’:
stopwatch.c:84:3: internal compiler error: Illegal instruction
   h  = (int) (sec / 3600.);
   ^
vectorops.c: In function ‘DLog’:
vectorops.c:242:5: internal compiler error: Illegal instruction
     else vec[x] = -DBL_MAX;
     ^~~~
alistat_main.c: In function ‘main’:
alistat_main.c:222:10: internal compiler error: Illegal instruction
          best * 100.,  msa->sqname[bestj],
          ^~~~
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please submit a full bug report,
with preprocessed source if appropriate.
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
Please include the complete backtrace with any bug report.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
See <https://gcc.gnu.org/bugs/> for instructions.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [stopwatch.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [vectorops.o] Error 1
make[1]: *** [alistat_main.o] Error 1
stopwatch.c: In function ‘format_time_string’:
stopwatch.c:84:3: internal compiler error: Illegal instruction
   h  = (int) (sec / 3600.);
   ^
vectorops.c: In function ‘DLog’:
vectorops.c:242:5: internal compiler error: Illegal instruction
     else vec[x] = -DBL_MAX;
     ^~~~
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [stopwatch.o] Error 1
make[1]: *** Waiting for unfinished jobs....
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [vectorops.o] Error 1
core_algorithms.c: In function ‘P7ViterbiTrace’:
core_algorithms.c:729:4: internal compiler error: Illegal instruction
    if (Prob2Score(hmm->begin[k+1], hmm->p1) + 1 * INTSCALE <= hmm->bsc[k+1])
    ^~
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [core_algorithms.o] Error 1
make[1]: *** Waiting for unfinished jobs....
display.c: In function ‘PrintIscore’:
display.c:335:3: internal compiler error: Illegal instruction
   div = INTSCALE / 0.693147180559945;   /* == INTSCALE / log(2) */
   ^~~
emulation.c: In function ‘WriteProfile’:
emulation.c:179:4: internal compiler error: Illegal instruction
    sc = sc * 100 / INTSCALE;
    ^~
core_algorithms.c: In function ‘P7ViterbiTrace’:
core_algorithms.c:729:4: internal compiler error: Illegal instruction
    if (Prob2Score(hmm->begin[k+1], hmm->p1) + 1 * INTSCALE <= hmm->bsc[k+1])
    ^~
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [display.o] Error 1
make[1]: *** Waiting for unfinished jobs....
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [emulation.o] Error 1
make[1]: *** [core_algorithms.o] Error 1
evd_test.c: In function ‘main’:
evd_test.c:109:3: internal compiler error: Illegal instruction
   mu         = -20.0;
   ^~
0x9d429f crash_signal
        ../.././gcc/toplev.c:337
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [evd_test.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
make failed: Inappropriate ioctl for device at lib/Tephra/Config/Install.pm line 319.

**Comment by sestaton** [comment]
Hi, 

I see a couple of issues. The first issue is that the Perl dependencies are not installed. Please see the [INSTALL](https://github.com/sestaton/tephra/blob/master/INSTALL.md) file for advice on installing all dependencies if you want to install from source.

Also, please let me know what operating system you are using. The compile errors are odd and might indicate an older version of gcc that has not been tested.

Thanks.

**Comment by yangxiaofeill** [comment]
I have install the Perl packages, and the system information is "Linux version 3.10.0-327.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.3 20140911 (Red Hat 4.8.3-9) (GCC) ) #1 SMP Thu Nov 19 22:10:57 UTC 2015"

**Comment by yangxiaofeill** [comment]
I have installed the software in another computer. However, when I run the following command:  `perl /opt/software/tephra/bin/tephra all -c tephra_config.yml`  
There are some error messages:
`INFO - ======== Tephra version: 0.12.2 (started at: 19-10-2018 15:36:27) ========
INFO - Configuration - Log file for monitoring progress and errors: scaffold_24.fafull.log
INFO - Configuration - Genome file:                                 scaffold_24.fa
INFO - Configuration - Repeat database:                             /home/xiaofei/annotation_dzj/annotation/annotation/Annotation_soft/maker/data/te_proteins.fasta
INFO - Configuration - Number of threads:                           24
INFO - Command - 'tephra findltrs' started at: 19-10-2018 15:36:27.
SSI index construction failed:
  primary keys not unique: 'Exo_endo_phos_2' occurs more than once
/root/.tephra/gt/bin/gt ltrdigest: error: Error executing hmmpress. Please make sure your pHMMs are valid andshare a common version (e.g. HMMER2 or HMMER3).
ERROR - LTRdigest failed with exit value: 1. Here is the output: 

INFO - Command - 'tephra findltrs' completed at: 19-10-2018 15:36:30.
INFO - Command - 'tephra findtrims' started at: 19-10-2018 15:36:30.
SSI index construction failed:
  primary keys not unique: 'Exo_endo_phos_2' occurs more than once
/root/.tephra/gt/bin/gt ltrdigest: error: Error executing hmmpress. Please make sure your pHMMs are valid andshare a common version (e.g. HMMER2 or HMMER3).
ERROR - LTRdigest failed with exit value: 1. Here is the output: 

INFO - Command - 'tephra findtrims' completed at: 19-10-2018 15:36:33.
INFO - Command - 'tephra findhelitrons' started at: 19-10-2018 15:36:33.

[WARNING]: Input FASTA is empty so the BLAST analysis will be skipped.

INFO - Command - 'tephra findhelitrons' completed at: 19-10-2018 15:36:39.
[fai_load] build FASTA index.
INFO - Output files - /home/xiaofei/acquilegia_anno/acoerulea_TE_tephra/scaffold_24_tephra_helitrons.gff3
INFO - Output files - /home/xiaofei/acquilegia_anno/acoerulea_TE_tephra/scaffold_24_tephra_helitrons.fasta
INFO - Command - 'tephra findtirs' started at: 19-10-2018 15:36:39.
ERROR - 'gt tirvish' failed exit value: 1. Here is the output: SSI index construction failed:
  primary keys not unique: 'Exo_endo_phos_2' occurs more than once
/root/.tephra/gt/bin/gt tirvish: error: Error executing hmmpress. Please make sure your pHMMs are valid andshare a common version (e.g. HMMER2 or HMMER3).


INFO - Command - 'tephra findtirs' completed at: 19-10-2018 15:36:42.
Use of uninitialized value $tirc_fas in -e at /root/perl5/lib/perl5/Tephra/Command/all.pm line 282.
INFO - Command - 'tephra findnonltrs' started at: 19-10-2018 15:36:42.

[WARNING]: No non-LTR elements were found on the forward strand. Will search reverse strand.


[WARNING]: No non-LTR elements were found on the reverse strand.


[WARNING]: No non-LTR elements were found so none will be reported.

INFO - Command - 'tephra findnonltrs' completed at: 19-10-2018 15:36:46.
INFO - Command - Generating combined FASTA file of complete elements at: 19-10-2018 15:36:46.
INFO - Results - Finished generating combined FASTA file of complete elements at: 19-10-2018 15:36:46. Final output files:
INFO - Output files - scaffold_24.fa_tephra_transposons_complete.fasta
INFO - Command - 'tephra maskref' for full-length transposon database started at: 19-10-2018 15:36:46.

[WARNING]: No bases were masked in 'scaffold_24.fa' under the specified conditions, so there will be no output.
Check the input, or try relaxing the alignment constraints if you believe matches should be found. Exiting.

INFO - Command - 'tephra maskref' completed at: 19-10-2018 15:36:49. Final output file:
Use of uninitialized value $final_mask in concatenation (.) or string at /root/perl5/lib/perl5/Tephra/Command/all.pm line 334.
INFO - Command - 'tephra findfragments' started at: 19-10-2018 15:36:49.

[ERROR]: Required arguments not given.

Name:
     tephra findfragments -  Search a masked genome with a repeat database to find fragmented elements

Description:
     This subcommand takes a database of full-length transposons and searches a masked genome to identify
     truncated or fragmented elements. The input database can be from Tephra or any other source. If the 'all' command
     is run then the results of this command will added to the final GFF3.

USAGE: tephra findfragments [-h] [-m]
    -m --man      :   Get the manual entry for a command.
    -h --help     :   Print the command usage.

Required:
    -g|genome     :   The masked genome sequences in FASTA format used to search for transposon fragments.
    -d|repeatdb   :   The file of repeat sequences in FASTA format used to query the genome.
    -o|outfile    :   The GFF3 file of non-overlapping transposon fragments.
    
Options:
    -t|threads    :   The number of threads to use for parallel BLAST searches (Default: 1).
    -p|percentid  :   The percent identity cutoff for BLAST alignments to the genome (Default: 80).
    -l|hitlen     :   The minimum length BLAST alignments to the genome (Default: 100).

INFO - Command - 'tephra findfragments' completed at: 19-10-2018 15:36:50.
INFO - Command - Generating combined FASTA file of fragments and complete elements at: 19-10-2018 15:36:50.`

It seems that the error was caused by HMMER software. I checked the PATH and added the HMMER software to the PATH, and also I found the HMMER software was installed  in /root/.tephra/.  The software list in /root/.tephra/ like: EMBOSS-6.5.7, gt, helitronscanner, hmm, hmmer-2.3.2, hmmer-3.1b2-linux-intel-x86_64, htslib-1.3.1, muscle, paml4.8, pHMM, TephraDB, vmatch.

**Comment by sestaton** [comment]
Okay, it is hard to follow now because it looks like two separate issues on two different machines. So we are on the same page, please go to the Tehpra source directory (the one you downloaded) and run:

    make test

And share the output. That will be more informative.

**Comment by sestaton** [comment]
Hi, please try the latest version (v0.12.3) if this is still of interest and then update if you have any problems.

There have been many code changes since this issue. Also, it would be helpful to show the output of 'make test' if there are any issues.

**Comment by sestaton** [comment]
I am going to close this due to no responses. If you try the latest version and find issues, please update this or open a new issue.
